Source/Parallelization/WarpXComm.cpp

```diff
@@ -55,23 +55,35 @@ WarpX::UpdateAuxilaryData ()
 {
     WARPX_PROFILE("WarpX::UpdateAuxilaryData()");
 
-    if (Bfield_aux[0][0]->ixType() == Bfield_fp[0][0]->ixType()) {
+    using ablastr::fields::Direction;
+
+    amrex::MultiFab *Bfield_aux_lvl0_0 = m_fields.get("Bfield_aux", Direction{0}, 0);
+
+    ablastr::fields::MultiLevelVectorField const& Bfield_fp = m_fields.get_mr_levels_alldirs("Bfield_fp", finest_level);
+
+    if (Bfield_aux_lvl0_0->ixType() == Bfield_fp[0][0]->ixType()) {
         UpdateAuxilaryDataSameType();
     } else {
         UpdateAuxilaryDataStagToNodal();
     }
 
     // When loading particle fields from file: add the external fields:
     for (int lev = 0; lev <= finest_level; ++lev) {
+        amrex::MultiFab *Bfield_aux_lvl_0 = m_fields.get("Bfield_aux", Direction{0}, lev);
+        amrex::MultiFab *Bfield_aux_lvl_1 = m_fields.get("Bfield_aux", Direction{1}, lev);
+        amrex::MultiFab *Bfield_aux_lvl_2 = m_fields.get("Bfield_aux", Direction{2}, lev);
+
         if (mypc->m_E_ext_particle_s == "read_from_file") {
-            amrex::MultiFab::Add(*Efield_aux[lev][0], *E_external_particle_field[lev][0], 0, 0, E_external_particle_field[lev][0]->nComp(), guard_cells.ng_FieldGather);
-            amrex::MultiFab::Add(*Efield_aux[lev][1], *E_external_particle_field[lev][1], 0, 0, E_external_particle_field[lev][1]->nComp(), guard_cells.ng_FieldGather);
-            amrex::MultiFab::Add(*Efield_aux[lev][2], *E_external_particle_field[lev][2], 0, 0, E_external_particle_field[lev][2]->nComp(), guard_cells.ng_FieldGather);
+            const auto& E_ext_lev = m_fields.get_alldirs("E_external_particle_field", lev);
+            amrex::MultiFab::Add(*Bfield_aux_lvl_0, *E_ext_lev[0], 0, 0, E_ext_lev[0]->nComp(), guard_cells.ng_FieldGather);
+            amrex::MultiFab::Add(*Bfield_aux_lvl_1, *E_ext_lev[1], 0, 0, E_ext_lev[1]->nComp(), guard_cells.ng_FieldGather);
+            amrex::MultiFab::Add(*Bfield_aux_lvl_2, *E_ext_lev[2], 0, 0, E_ext_lev[2]->nComp(), guard_cells.ng_FieldGather);
```

Comment:
For those looking into tests with external fields, I think these should be `Efield_aux` variables. However, they need to be added above properly (renaming just here won't compile). Right now I'm trying to track another bug, so I'll get to this later one, unless someone pushes the fix first.

Reply:
@dpgrote 
The external fields tests, all in RZ geometry, pass despite the bug reported here. Isn't that strange?

Reply:
Pinging @RemiLehe and @ax3l on this question too, just to make sure it does not get lost.

Reply:
Thanks! Fixed via cf76e6a09a51a0d7baeac2feec8150cc3a70ae4d